test/jdk/javax/swing/JButton/bug4323121.java

```diff
@@ -30,96 +30,100 @@
  * @run main bug4323121
  */
 
-import java.awt.Graphics;
 import java.awt.Point;
 import java.awt.Robot;
+import java.awt.event.MouseAdapter;
 import java.awt.event.MouseEvent;
-import java.awt.event.MouseListener;
-import java.awt.event.MouseMotionListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.util.concurrent.CountDownLatch;
+
 import javax.swing.JButton;
 import javax.swing.JFrame;
 import javax.swing.SwingUtilities;
 
-public class bug4323121 {
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public final class bug4323121 {
 
     static JFrame frame;
-    static testButton button;
-    static volatile Point pt;
-    static volatile int buttonW;
-    static volatile int buttonH;
-    static volatile boolean failed = false;
+    static JButton button;
+
+    static volatile Point buttonCenter;
+
+    private static final CountDownLatch windowGainedFocus = new CountDownLatch(1);
+
+    private static final CountDownLatch mouseEntered = new CountDownLatch(1);
+
+    // Usage of this flag is thread-safe because of using the mouseEntered latch
+    private static boolean modelArmed;
```

Comment:
should it be volatile?

Reply:
No, it shouldn't: _“this flag is thread-safe because of using the `mouseEntered` latch”_.

Reply:
Ok.